src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -982,6 +982,8 @@ class Config:
                     ad = await self.config()
 
             ret = neterr.JOINED
+            await self._net_ads_setspn(f'nfs/{ad["netbiosname"]}.{ad[domainname"}')
+            await self._net_ads_setspn(f'nfs/{ad["netbiosname"]}')
```

Comment:
Check return value for these and generate an alert on failure?

Reply:
I moved this up to the code block where we're performing keytab related tasks on non-legacy AD configurations. Change of trust pw is performed automatically, and resulting keytab uploaded to FN config.